### ISSUE TITLE
Support implicit generic type aliases that use `type[]`

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#4441]
### Summary

`f` and `g` here should be inferred the same, but they currently are not:

```py
from typing import TypeVar

DecoratorT = TypeVar("DecoratorT", bound=object)
DecoratedClass = type[DecoratorT]

def f(x: type[DecoratorT]) -> type[DecoratorT]:
    return x

def g(x: DecoratedClass) -> DecoratedClass:
    return x

class Foo: ...

reveal_type(f(Foo))  # type[Foo]
reveal_type(g(Foo))  # type[Unknown]
```

That means that [this test](https://github.com/astral-sh/ruff/blob/597d2daf81668b1fb97de671c8e7bad0269c1e58/crates/ty_python_semantic/resources/mdtest/decorators.md?plain=1#L725-L757) is not testing what it thinks it's testing -- the decorator is silently returning `type[Unknown]`. (I will clean that up as part of https://github.com/astral-sh/ruff/pull/28128, though)

**Comment by jelle-openai** [comment]
I actually think ty's current behavior is correct here. If you use just `DecoratedClass`, you're using a generic alias without specializing it, and it implicitly specializes to Any. All other type checkers (except pycroscope) also behave the way ty currently behaves.

**Comment by AlexWaygood** [comment]
oh yeah, you're right
